### PR TITLE
Scope validation for endpoints

### PR DESCRIPTION
**Pull request by gkallenberg**
Add basic scopes and validation for the various endpoints. Parses the header and examines the OAuth2 scopes and allows/denies based on the client configuration passed.

### CODE REVIEW

**Review by rhernand3z — APPROVED** [review]
Looks good.

File: src/ServiceController.php [hunk 1]
```diff
@@ -11,6 +12,12 @@
  */
 class ServiceController extends Controller
 {
+    const READ_REQUEST_SCOPE = 'read:holds';
```
Comment (kfriedman) [inline src/ServiceController.php]: Are these are `holdrequests`? I think a `Hold` is what happens after a `HoldRequest` is successful. I'm just thinking ahead to when we actually have a `HoldsService`.

File: src/Controller/HoldRequestController.php [hunk 2]
```diff
@@ -45,34 +46,49 @@ class HoldRequestController extends ServiceController
...
      *     ),
      *     @SWG\Response(
      *         response="500",
      *         description="Generic server error",
-     *         @SWG\Schema(ref="#/definitions/ErrorResponse")
+     *         @SWG\Schema(ref="#/definitions/HoldRequestErrorResponse")
      *     ),
      *     security={
      *         {
      *             "api_auth": {"openid offline_access api"}
```
Comment (kfriedman) [inline src/Controller/HoldRequestController.php]: You might want to update the documentation to include the actual scopes that are required. This is finally becoming meaningful! 😄

**Review by kfriedman — COMMENTED** [review]
Looks great. A couple minor questions/comments. Nice work!

**Comment by gkallenberg** [comment]
I'll fix the scopes as per our earlier conversation to hold-requests or
similar and update the README.

--

*Gregory Kallenberg* | The New York Public Library
*Senior Applications Developer*

Digital Development
40 West 20th Street, 5th Floor, New York, NY 10011
T: 212.621.0283 | gregorykallenberg@nypl.org
<http://www.nypl.org>
*Inspiring Lifelong Learning | Advancing Knowledge | Strengthening Our
Communities*

**Comment by gkallenberg** [comment]
Fixed the scopes according to the specification document, and updated the swagger docs in the controller.

**Review by kfriedman — APPROVED** [review]

**Pull request merged**